### PR TITLE
Add custom rally plugin 

### PR DESCRIPTION
**Pull request by SwanDr**
Add parameter on run.yml to enable the use of custom rally plugin
Add example of custom plugin with the associated scenario

### CODE REVIEW

File: workload/run.yml [hunk 1]
```diff
@@ -11,10 +11,12 @@ rally:
...
-      - name: boot and list servers 
-        file: nova-boot-list-cc.yml
-        args: 
-          sla_max_avg_duration: 30
+    - name: boot and list servers
+      file: scenarios/nova-boot-list-cc.yml
+      args:
+        sla_max_avg_duration: 30
+#If you use custom rally plugin provide the path here
+      #plugin: plugins/your_plugin
```
Comment (msimonin) [inline workload/run.yml]: I would rephrase :

```
#  plugin: path to your plugin relative to the workload dir
```

File: enos/ansible/roles/bench/tasks/rally.yml [hunk 2]
```diff
@@ -38,21 +38,32 @@
     command: rally deployment create --fromenv --name=discovery
 
 - name: Copy rally scenarios
-  copy: src={{ bench.location }} dest=/root/rally_home/ owner=655500
+  copy: src={{ bench.scenario_location }} dest=/root/rally_home/ owner=655500
+
+- name: Create plugin directory
+  when: bench.plugin_location is defined
```
Comment (msimonin) [inline enos/ansible/roles/bench/tasks/rally.yml]: Maybe put the when at the end of the task to be consistent with the following tasks

File: workload/run.yml [hunk 3]
```diff
@@ -11,10 +11,12 @@ rally:
       times: 
         - 10
     scenarios:
-      - name: boot and list servers 
-        file: nova-boot-list-cc.yml
-        args: 
-          sla_max_avg_duration: 30
+    - name: boot and list servers
+      file: scenarios/nova-boot-list-cc.yml
```
Comment (msimonin) [inline workload/run.yml]: Since the workload directory is only an example, I would prefer not to create a subdirectory only for one file.

File: workload/run.yml (hunk 1 above)
Comment (msimonin) [inline workload/run.yml]: Note that we could also pass a directory :)

File: workload/plugins/custom-plugin-select-host-and-boot.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+from rally.plugins.openstack import scenario
```
Comment (SwanDr) [inline workload/plugins/custom-plugin-select-host-and-boot.py]: I must to import this file it's useful for: https://github.com/BeyondTheClouds/enos/pull/141/files#diff-92e32cbb6ed56a30fea47aa5fcceefe8R31

File: reservation.yaml.sample [hunk 5]
```diff
@@ -3,26 +3,32 @@
 # Enos Configuration File                         #
```
Comment (msimonin) [inline reservation.yaml.sample]: I would prefer to not modify this file

Comment (msimonin) [inline reservation.yaml.sample]: I would prefer not to modify this file.

File: workload/plugins/custom-plugin-select-host-and-boot.py (hunk 4 above)
Comment (msimonin) [inline workload/plugins/custom-plugin-select-host-and-boot.py]: It's maybe better to keep all your files separated from the master.

I suggest you to create a scenario dedicated to your use case : see https://github.com/BeyondTheClouds/enos-scenarios

If you want you can push your configuration there : e.g the workload dir

File: workload/run.yml [hunk 6]
```diff
@@ -11,10 +11,11 @@ rally:
       times: 
         - 10
     scenarios:
-      - name: boot and list servers 
-        file: nova-boot-list-cc.yml
-        args: 
-          sla_max_avg_duration: 30
+    - name: boot and list
```
Comment (msimonin) [inline workload/run.yml]: indentation looks weird here

File: workload/select-host-and-boot.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (msimonin) [inline workload/select-host-and-boot.yml]: put this file in a dedicated enos-scenario.

File: reservation.yaml.sample [hunk 8]
```diff
@@ -56,3 +58,4 @@ kolla:
   kolla_install_type: "source"
   docker_namespace: "beyondtheclouds"
   enable_heat: "no"
+  neutron_plugin_agent: "linuxbridge"
```
Comment (msimonin) [inline reservation.yaml.sample]: you can push this in a specific enos-scenario (see after)

File: workload/plugins/custom-plugin-select-host-and-boot.py (hunk 4 above)
Comment (SwanDr) [inline workload/plugins/custom-plugin-select-host-and-boot.py]: ok
